BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/SelfDiagnostics/SelfDiagnosticsConfigParser.cs

```diff
@@ -31,11 +36,149 @@ internal class SelfDiagnosticsConfigParser
+            {
+                return false;
+            }
+
+            UpdateFileSizeToBeWithinLimit(ref fileSizeInKB);
+
+            if (!TryParseLogLevel(ParseLocation.EnviornmentVariable, Environment.GetEnvironmentVariable(LogLevel), out logLevel))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        internal static bool TryParseLogDirectory(ParseLocation location, string val, out string logDirectory)
+        {
+            logDirectory = null;
+            if (location == ParseLocation.EnviornmentVariable)
+            {
+                if (!String.IsNullOrWhiteSpace(val))
+                {
+                    logDirectory = val;
+                    return true;
+                }
+
+                // Short circuit for this parse from enviornment variables path
+                // to check whether self diagnostics feature was enabled by the json config file.
+                return false;
+            }
+            else 
+            {
+                var logDirectoryResult = LogDirectoryRegex.Match(val);
+                logDirectory = logDirectoryResult.Groups[LogDirectory].Value;
+                return logDirectoryResult.Success && !string.IsNullOrWhiteSpace(logDirectory);
+            }
+        }
+
+        internal static bool TryParseFileSize(ParseLocation location, string val, out int fileSizeInKB)
+        {
+            fileSizeInKB = FileSizeLowerLimit;
+            if (location == ParseLocation.EnviornmentVariable)
+            {
+                if (!String.IsNullOrEmpty(val))
+                {
+                    return int.TryParse(val, out fileSizeInKB);
+                }
+
+                // If the LogDirectory was set by the environment variable,
+                // but FileSize was not set, use the default fileSize.
+                return true;
+            }
+            else
+            {
+                var fileSizeResult = FileSizeRegex.Match(val);
+                return fileSizeResult.Success && int.TryParse(fileSizeResult.Groups[FileSize].Value, out fileSizeInKB);
+            }
+        }
+
+        // no-op if the fileSize is within range
+        internal static void UpdateFileSizeToBeWithinLimit(ref int fileSizeInKB) 
+        {
+            if (fileSizeInKB < FileSizeLowerLimit)
+            {
+                fileSizeInKB = FileSizeLowerLimit;
+                return;
+            }
+            
+            if (fileSizeInKB > FileSizeUpperLimit)
+            {
+                fileSizeInKB = FileSizeUpperLimit;
+            }
+        }
+
+        internal static bool TryParseLogLevel(ParseLocation location, string val, out EventLevel logLevel)
+        {
+            logLevel = EventLevel.LogAlways;
+            if (location == ParseLocation.EnviornmentVariable)
+            {
+                if (!String.IsNullOrEmpty(val))
+                {
+                    logLevel = (EventLevel)Enum.Parse(typeof(EventLevel), val);
```

Comment:
What will happen if `val` has invalid value?
Should we consider using `TryParse`?